analysea/tide.py

```diff
@@ -113,86 +102,107 @@ def demean_amps_phases(
     return const, mean_amps, mean_phases  # ignore mypy
 
 
-def calc_constituents(ts: pd.Series, kwargs: Dict[str, Any]) -> Any:
+def calc_constituents(
+    ts: pd.Series, resample_time: int = 10, lat: float = 0.0, **kwargs: Optional[Dict[str, Any]]
+) -> Any:
     # resample to 10 min for a MUCH faster analysis
     # https://github.com/wesleybowman/UTide/issues/103
-    h_rsmp = ts.resample("10min").mean()
+    h_rsmp = ts.resample(f"{resample_time}min").mean()
+
     ts = h_rsmp.index
-    coef = utide.solve(ts, h_rsmp, **kwargs)
+    # shifting half the sampling time
+    # > to recenter the averaged signal
+    df = h_rsmp.shift(freq=f"{resample_time // 2}min")
+    ts = df.index
+    df = h_rsmp.iloc[:, 0].values
```

Comment:
Why do we need to make a copy? Does the object get mutated?
Also why do we convert to a numpy array. Is it faster?